packages/labs/preact-signals/README.md

```diff
@@ -0,0 +1,127 @@
+
+This effectively makes the the return result of `render()` a computed signal.
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit';
+import {SignalWatcher, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count.value}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.value = count.value + 1;
+  }
+}
+```
+
+Elements should not _write_ to signals in these lifecycle methods or they might cause an infinite loop.
+
+### watch() directive
+
+The `watch()` directive accepts a single Signal and renders its value, subscribing to updates and updating the DOM when the signal changes.
+
+The `watch()` directive allows for very targeted updates of the DOM, which can be good for performance (but as always, measure!). The downside is that the lifecycle callbacks are not automatically watched for signal access, so values computed from signals must by wrapped in computed signals.
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit';
+import {watch, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends LitElement {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${watch(count)}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.value = count.value + 1;
+  }
+}
+```
+
+You can mix and match the `SignalWatcher` mixins and the `watch()` directive. When you pass a signal directly to `watch()` it is not accessed in a callback watched by `SignalWatcher`, so an update to that signal will cause a targeted DOM update and not an entire element update.
+
+### html tag and withWatch()
+
+This package also exports an `html` template tag that can be used in place of Lit's default `html` tag and automatically wraps any signals in `watch()`.
+
+```ts
+import {LitElement} from 'lit';
+import {customElement, property} from 'lit';
+import {html, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends LitElement {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
```

Comment:
Nit: button button button button.

But seriously, this is an amazing library to provide! I can't wait to use it!